### ISSUE TITLE
get_evaluation_parameter_dependencies raises KeyError when evaluating  dynamic query Evaluation parameters via Expectation suites

### ISSUE THREAD

**Issue by LavinaVRovine** [fivetran/great_expectations#2795]
**Describe the bug**

get_evaluation_parameter_dependencies raises KeyError when evaluating dynamic query Evaluation parameters via Expectation suites.

Following https://docs.greatexpectations.io/en/latest/guides/how_to_guides/creating_and_editing_expectations/how_to_dynamically_load_evaluation_parameters_from_a_database.html

setup works, only when NOT validating via expectation suite.

**To Reproduce**
Run the setup from the link above via expectation suites:
1. Define the suite in my_suite.json as

```
{
  "data_asset_type": "Dataset",
  "expectation_suite_name": "my_suite",
  "expectations": [

      {
      "expectation_type": "expect_column_values_to_be_in_set",
      "kwargs": {
        "column": "a",
        "value_set": {
          "$PARAMETER": "urn:great_expectations:stores:my_query_store:current_genre_ids"
        }
      },
      "meta": {}
    }
  ],
  "meta": {

  }
}
```

2. Define the checkpoint or create batch in any other way. My Checkpoint in my_check.yml 
``` validation_operator_name: action_list_operator

batches:
  - batch_kwargs:
      query: "SELECT a from fake_tmp"
      datasource: logex_data_warehouse
      data_asset_name: cmoon
    expectation_suite_names: # one or more suites may validate against a single batch
      - my_suite
```
3. Run validation. My setup:
``` 
chk = context.get_checkpoint("my_check")
chk.run()
```
```
Traceback (most recent call last):
  File "C:\Users\pavel.klammert\Miniconda3\envs\uk_validation_dash\lib\site-packages\great_expectations\validation_operators\validation_operators.py", line 388, in _run_actions
    action_result = self.actions[action["name"]].run(
  File "C:\Users\pavel.klammert\Miniconda3\envs\uk_validation_dash\lib\site-packages\great_expectations\checkpoint\actions.py", line 58, in run
    return self._run(
  File "C:\Users\pavel.klammert\Miniconda3\envs\uk_validation_dash\lib\site-packages\great_expectations\checkpoint\actions.py", line 789, in _run
    self.data_context.store_evaluation_parameters(validation_result_suite)
  File "C:\Users\pavel.klammert\Miniconda3\envs\uk_validation_dash\lib\site-packages\great_expectations\data_context\data_context.py", line 2104, in store_evaluation_parameters
    self._compile_evaluation_parameter_dependencies()
  File "C:\Users\pavel.klammert\Miniconda3\envs\uk_validation_dash\lib\site-packages\great_expectations\data_context\data_context.py", line 2140, in _compile_evaluation_parameter_dependencies
    dependencies = expectation_suite.get_evaluation_parameter_dependencies()
  File "C:\Users\pavel.klammert\Miniconda3\envs\uk_validation_dash\lib\site-packages\great_expectations\core\expectation_suite.py", line 173, in get_evaluation_parameter_dependencies
    t = expectation.get_evaluation_parameter_dependencies()
  File "C:\Users\pavel.klammert\Miniconda3\envs\uk_validation_dash\lib\site-packages\great_expectations\core\expectation_configuration.py", line 1210, in get_evaluation_parameter_dependencies
    {urn["expectation_suite_name"]: [urn["metric_name"]]},
  File "C:\Users\pavel.klammert\Miniconda3\envs\uk_validation_dash\lib\site-packages\pyparsing.py", line 598, in __getitem__
    return self.__tokdict[i][-1][0]
KeyError: 'expectation_suite_name'
Traceback (most recent call last):
  File "C:\Users\pavel.klammert\Miniconda3\envs\uk_validation_dash\lib\site-packages\great_expectations\core\usage_statistics\usage_statistics.py", line 262, in usage_statistics_wrapped_method
    result = func(*args, **kwargs)
  File "C:\Users\pavel.klammert\Miniconda3\envs\uk_validation_dash\lib\site-packages\great_expectations\data_context\data_context.py", line 1284, in run_validation_operator
    return validation_operator.run(
  File "C:\Users\pavel.klammert\Miniconda3\envs\uk_validation_dash\lib\site-packages\great_expectations\validation_operators\validation_operators.py", line 332, in run
    batch_actions_results = self._run_actions(
  File "C:\Users\pavel.klammert\Miniconda3\envs\uk_validation_dash\lib\site-packages\great_expectations\validation_operators\validation_operators.py", line 407, in _run_actions
    raise e
  File "C:\Users\pavel.klammert\Miniconda3\envs\uk_validation_dash\lib\site-packages\great_expectations\validation_operators\validation_operators.py", line 388, in _run_actions
    action_result = self.actions[action["name"]].run(
  File "C:\Users\pavel.klammert\Miniconda3\envs\uk_validation_dash\lib\site-packages\great_expectations\checkpoint\actions.py", line 58, in run
    return self._run(
  File "C:\Users\pavel.klammert\Miniconda3\envs\uk_validation_dash\lib\site-packages\great_expectations\checkpoint\actions.py", line 789, in _run
    self.data_context.store_evaluation_parameters(validation_result_suite)
  File "C:\Users\pavel.klammert\Miniconda3\envs\uk_validation_dash\lib\site-packages\great_expectations\data_context\data_context.py", line 2104, in store_evaluation_parameters
    self._compile_evaluation_parameter_dependencies()
  File "C:\Users\pavel.klammert\Miniconda3\envs\uk_validation_dash\lib\site-packages\great_expectations\data_context\data_context.py", line 2140, in _compile_evaluation_parameter_dependencies
    dependencies = expectation_suite.get_evaluation_parameter_dependencies()
  File "C:\Users\pavel.klammert\Miniconda3\envs\uk_validation_dash\lib\site-packages\great_expectations\core\expectation_suite.py", line 173, in get_evaluation_parameter_dependencies
    t = expectation.get_evaluation_parameter_dependencies()
  File "C:\Users\pavel.klammert\Miniconda3\envs\uk_validation_dash\lib\site-packages\great_expectations\core\expectation_configuration.py", line 1210, in get_evaluation_parameter_dependencies
    {urn["expectation_suite_name"]: [urn["metric_name"]]},
  File "C:\Users\pavel.klammert\Miniconda3\envs\uk_validation_dash\lib\site-packages\pyparsing.py", line 598, in __getitem__
    return self.__tokdict[i][-1][0]
KeyError: 'expectation_suite_name'
```

**Additional context**

During validation, the requested key is correctly obtained from the store via evaluation_parameters.py.parse_evaluation_parameter(line-308)

However, later the expectation_configuration.get_evaluation_parameter_dependencies fails, as it does not consider there might be a store type urn; It probably expects to have a dependency on another expectation only.

This simple addition to expectation_configuration.get_evaluation_parameter_dependencies:
```
...
urns = parsed_dependencies.get("urns", [])
for string_urn in urns:
    try:
        urn = ge_urn.parseString(string_urn)
    except ParserError:
        logger.warning(
            "Unable to parse great_expectations urn {}".format(
                value["$PARAMETER"]
            )
        )
        continue
    # addition
    if urn["urn_type"] == "stores":
        continue
    # addition end
    if not urn.get("metric_kwargs"):
        nested_update(
            dependencies,
            {urn["expectation_suite_name"]: [urn["metric_name"]]},
        )
    else: ...
```
Solves the issue, but as I do not have any real understanding of get_evaluation_parameter_dependencies, I have no idea of the implications of this simple fix.
Also please note, the evaluated reference in data docs will not be shown when doing the above (no idea whether it's due to the update or not), only the "$PLACEHOLDER" flag.

**Environment (please complete the following information):**
 - Operating System: Windows
 - Great Expectations Version: 0.13.12

**Comment by eugmandel** [comment]
@LavinaVRovine Thank you - we will look into this issue.

**Comment by DanielSastoque** [comment]
Hi! Any update on this one? I'm having the same issue. Is there an alternative in the meantime? 

thanks!

**Comment by LavinaVRovine** [comment]
Hi @DanielSastoque ,

can't comment on the fix.
In a meantime, I just resolve evaluation parameters explicitly and pass them as evaluation parameters. code:

![obrazek](https://user-images.githubusercontent.com/8156523/136183583-9831b916-1464-47be-9a9b-c79b3891d4c2.png)

**Comment by talagluck** [comment]
Hi @LavinaVRovine and @DanielSastoque - thanks so much for reporting this, and apologies for delays. 

We merged a fix (#3580) for another Evaluation Parameters issue yesterday, and I wonder if it may resolve this as well. Are you able to run Great Expectations from `develop` and see if this works for you? Otherwise we will take another look.

**Comment by talagluck** [comment]
Hi @LavinaVRovine and @DanielSastoque, I'm going to close this for now, but if this does turn out to still be an issue, please feel free to re-open.
